commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -0,0 +1,161 @@
+        getQueue().add(request);
+    }
+
+    private RequestQueue getQueue() {
+        return getQueue(context);
+    }
+
+    private static RequestQueue getQueue(Context context) {
+        if (REQUEST_QUEUE == null) {
+            REQUEST_QUEUE = Volley.newRequestQueue(context.getApplicationContext());
+        }
+        return REQUEST_QUEUE;
+    }
+
+    private static class LogResponseListener<T> implements Response.Listener<T> {
+        private static final String TAG = LogResponseListener.class.getName();
+
+        @Override
+        public void onResponse(T response) {
+            Log.d(TAG, response.toString());
+        }
+    }
+
+    private static class LogResponseErrorListener implements Response.ErrorListener {
+        private static final String TAG = LogResponseErrorListener.class.getName();
+
+        @Override
+        public void onErrorResponse(VolleyError error) {
+            Log.e(TAG, error.toString());
+        }
+    }
+
+    private static class QueryRequest extends JsonRequest<QueryResponse> {
+        private static final String TAG = QueryRequest.class.getName();
+
+        public QueryRequest(String url,
+                            Response.Listener<QueryResponse> listener,
+                            Response.ErrorListener errorListener) {
+            super(Request.Method.GET, url, null, listener, errorListener);
+        }
+
+        @Override
+        protected Response<QueryResponse> parseNetworkResponse(NetworkResponse response) {
+            String json = parseString(response);
+            //Log.d(TAG, "json=" + json);
+            QueryResponse queryResponse = GSON.fromJson(json, QueryResponse.class);
+            return Response.success(queryResponse, cacheEntry(response));
+        }
+
+
+        private Cache.Entry cacheEntry(NetworkResponse response) {
+            return HttpHeaderParser.parseCacheHeaders(response);
+        }
+
+
+        private String parseString(NetworkResponse response) {
+            try {
+                return new String(response.data, HttpHeaderParser.parseCharset(response.headers));
+            } catch (UnsupportedEncodingException e) {
+                return new String(response.data);
+            }
+        }
+    }
+
+    private static class QueryResponse {
+        private Query query;
+
+        @Override
+        public String toString() {
+            return "query=" + query.toString();
+        }
+    }
+
+    private static class Query {
+        private Page [] pages;
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder("pages=" + "\n");
+            for (Page page : pages) {
+                builder.append(page.toString());
+                builder.append("\n");
+            }
+            builder.replace(builder.length() - 1, builder.length(), "");
+            return builder.toString();
+        }
+    }
+
+    private static class Page {
```

Comment:
This class seems like it should exist in its own file. It's a model that might have usage elsewhere. If it doesn't, it could still be in its own file but package protected.